src/services/branch-manager.js

```diff
@@ -42,6 +43,16 @@ function createBranch(branchName, environmentSecret) {
     .send({ branchName: encodeURIComponent(branchName) });
 }
 
+function switchBranch({ id }, environmentSecret) {
+  const authToken = authenticator.getAuthToken();
+
+  return agent
+    .put(`${serverHost()}/api/environments`)
```

Comment:
Call on environment route inside `BranchManager` is pretty strange no ?

Reply:
We could consider it valid, however I'm OK to change my mind! 

The actual action is related to branch: the action is indeed **`switch branch`**, so it could be in the **branch service**. The `switchBranch` function have to **patch** an environment to update its current branch.

When I look at environment service, I see:

- /api/projects/xxx//environments
- /api/environments/
- /api/deployment-requests/

So it's not consistent either with the service name. I don't know if it's an issue.

Still, I do not have a real preference: I guess it could be in environment service too. So, if you have a preference, I would follow yours! It's up to you